### ISSUE TITLE
Encountered file(s) that should have been pointers during clone

### ISSUE THREAD

**Issue by mloskot** [git-lfs/git-lfs#3389]
I've used Git for Windows 2.19 with Git LFS 2.6.0 to migrate a repo to Git LFS as follows, in Git Bash: 

```
git lfs migrate import --everything --verbose --include="*.exe,*.dll,*.lib,*.pdb"
```

My `~/.gitconfig` file has this LFS settings:

```
[filter "lfs"]
    clean = git-lfs clean -- %f
    smudge = git-lfs smudge --skip -- %f
    process = git-lfs filter-process --skip
    required = true
```

I run a test clone

```
git clone repo.git
Cloning into 'repo'...
remote: Enumerating objects: 1315613, done.
remote: Total 1315613 (delta 0), reused 0 (delta 0), pack-reused 1315613
Receiving objects: 100% (1315613/1315613), 2.07 GiB | 10.21 MiB/s, done.
Resolving deltas: 100% (497096/497096), done.
Checking out files: 100% (229742/229742), done.
Encountered 6 file(s) that should have been pointers, but weren't:
        ThirdParty/mingw64/share/licenses/gcc-libs/COPYING.LIB
        ThirdParty/mingw64/share/licenses/gettext/gettext-runtime/intl/COPYING.LIB
        ThirdParty/mingw64/share/licenses/gettext/gettext-runtime/libasprintf/COPYING.LIB
        ThirdParty/mingw64/share/licenses/libiconv/COPYING.LIB
        ThirdParty/mingw64/share/licenses/libiconv/libcharset/COPYING.LIB
        ThirdParty/mingw64/share/licenses/libtasn1/LICENSE.LIB
```

Since the LFS tracking seems to be in-place for `.lib` files

```
cat .gitattributes | grep .lib
*.lib filter=lfs diff=lfs merge=lfs -text
```

why the `git clone` complains about `COPYING.LIB` files?

I can see that this is tricky, because:

- `COPYING.LIB` is in fact a small text file
- `*.LIB` may not necessarily be the same as `*.lib`

However, why the 'warning' and how should I deal with it, if at all?

-----

I don't think my issue is related to seemingly similar #1939 or #2910.  The former speaks about the issue after `git checkout`. The latter about "the clean/smudge mechanism", but in my `git clone` case no clean applies, and smudge is skipped. Am I correct?

**Comment by bk2204** [comment]
When Git LFS knows you're on a case-insensitive file system, it matches patterns case insensitively.  So on Windows, `*.lib` and `*.LIB` are identical.  Because Git LFS thinks those files are supposed to be handled by it, but they're not, it complains.

If you don't want to match `COPYING.LIB` files, the easiest way is to do something like the following in your `.gitattributes` file:

```
*.lib filter=lfs diff=lfs merge=lfs -text
COPYING.LIB -filter diff merge text
```

That will set `COPYING.LIB` files to be handled as plain Git files and not as LFS files.

**Comment by mloskot** [comment]
> on Windows, *.lib and *.LIB are identical.

Yes.
And, assuming I migrated and cloned on Windows, in Git Bash, `COPYING.LIB` should remain tracked as Git LFS - as per `*.lib` in my `.gitattributes`. But, for some reason it is not and clone issues `Encountered 6 file(s) that should have been pointers, but weren't'.

What am I missing in my reasoning?

**Comment by bk2204** [comment]
Ah, good catch.  I believe this is actually an issue I first noticed the other day when browsing around the `wildmatch` package.  I think our internal matching code is backwards; that is, it is case insensitive on Linux, and case sensitive on Mac and Windows.  When we do the filtering on Windows, we match case sensitively, but Git matches insensitively, and so we complain.

I'll try to work up a fix.

**Comment by mloskot** [comment]
That explains it indeed! Thank you, and for the upcoming fix too.

**Comment by bk2204** [comment]
Just FYI, this is going to be fixed in git-lfs/wildmatch#11.
